pycon/settings.py

```diff
@@ -1,1266 +1,1382 @@
-# have to configure your proxy to send proper X-Forward-* headers and enable
-# SECURE_PROXY_SSL_HEADER.
-#
-if not DEBUG:
-    # Only set this in production mode, since debug servers typically don't
-    # have HTTPS set up.
-    os.environ['HTTPS'] = 'on'
-    # HTTPS configuration
-    HTTPS = True
-    SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
-else:
-    # Turn off HTTPS
-    if 'HTTPS' in os.environ:
-        del os.environ['HTTPS']
-    HTTPS = False
+from configurations import Configuration, values, pristinemethod
+from model_utils import Choices
 
+# TODO: Clarification on the translation status required.
 #from django.utils.translation import ugettext as _
 _ = lambda x:x
 
-ADMINS = (
-    ('web-wg', 'web-wg@europython.eu'),
-)
-
-MANAGERS = ADMINS
-
-ALLOWED_HOSTS = os.environ.get("ALLOWED_HOSTS", '*').split(',')
-#APPEND_SLASH = False
 
-PROJECT_DIR = os.environ.get('PROJECT_DIR', os.path.normpath(
-    os.path.join(os.path.dirname(__file__), '..')))
-DATA_DIR = os.environ.get('DATA_DIR', os.path.join(PROJECT_DIR, 'data'))
-OTHER_STUFF = os.environ.get('OTHER_STUFF',
-                             os.path.join(PROJECT_DIR, 'documents'))
+class Base(Configuration):
+    DEBUG = values.BooleanValue(False)
 
-LOGS_DIR = os.path.join(PROJECT_DIR, 'logs/')
-
-if not os.path.exists(LOGS_DIR):
-    os.makedirs(LOGS_DIR)
-
-sys.path.insert(0, os.path.join(PROJECT_DIR, 'deps'))
-
-SITE_DATA_ROOT = DATA_DIR + '/site'
-
-DATABASE_TYPE = os.environ.get('DATABASE')
+    # We want to use HTTPS for everything and not fiddle with docker or gunicorn
+    # setups.
+    #
+    # See http://security.stackexchange.com/questions/8964/trying-to-make-a-django-based-s
+    # for details.
+    #
+    # Note: This doesn't really help all that much. In order to Django behave, you
+    # have to configure your proxy to send proper X-Forward-* headers and enable
+    # SECURE_PROXY_SSL_HEADER.
+    #
+    if not DEBUG:
+        # Only set this in production mode, since debug servers typically don't
+        # have HTTPS set up.
+        os.environ['HTTPS'] = 'on'
+        # HTTPS configuration
+        HTTPS = True
+        SECURE_PROXY_SSL_HEADER = ('HTTP_X_FORWARDED_PROTO', 'https')
+    else:
+        # Turn off HTTPS
+        if 'HTTPS' in os.environ:
+            del os.environ['HTTPS']
+        HTTPS = False
+
+
+    ADMINS = (
+        ('web-wg', 'web-wg@europython.eu'),
+    )
+    MANAGERS = ADMINS
+
+    #APPEND_SLASH = False
+
+    @property
+    def ALLOWED_HOSTS(self):
+        return self.ALLOWED_HOSTS_FROM_ENV.split(',')
+    ALLOWED_HOSTS_FROM_ENV = values.Value('*', environ_name='ALLOWED_HOSTS')
+
+    # Project directories settings.
+    PROJECT_DIR = os.path.normpath(os.path.join(os.path.dirname(__file__), '..'))
+    DATA_DIR = values.Value(os.path.join(PROJECT_DIR, 'data'))
+    OTHER_STUFF = values.Value(os.path.join(PROJECT_DIR, 'documents'))
+    LOGS_DIR = os.path.join(PROJECT_DIR, 'logs/')
+
+    @property
```

Comment:
why `@property` instead of simple `SITE_DATA_ROOT = DATA_DIR + '/site'`?

Reply:
Properties are used for calculated settings - it's the library's way of avoid race conditions. More details here: https://django-configurations.readthedocs.io/en/stable/patterns/#property-settings.